app/models/spree/gateway/braintree_gateway_decorator.rb

```diff
@@ -45,10 +47,14 @@ def authorize money, credit_card, options={}
       # Include device data if present
       options[:device_data] = credit_card.device_data
 
+      current_merchant_account_id = BraintreePresenter.new(credit_card.payment_method).merchant_account_id(money)
+      options[:merchant_account_id] = current_merchant_account_id
+      credit_card.payment_method.preferences[:merchant_account_id] = current_merchant_account_id
```

Comment:
Saw this line and was perplexed by it as well. This is updating the global Spree preference. First it doesn't appear this preference is registered. Second nothing uses that preference, it would be subject to race conditions and it generates unnecessary SQL. My suggestion is replace all three lines with:

```
options[:merchant_account_id] = BraintreePresenter.new(credit_card.payment_method).merchant_account_id(money)
```

Wanted to see if I could get more info before I did that cleanup.